src/canvas-extractor/edfi_canvas_extractor/graphql/students.py

```diff
@@ -0,0 +1,77 @@
+﻿# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from typing import List
+
+from pandas import DataFrame
+import sqlalchemy
+
+from canvasapi.user import User
+from edfi_lms_extractor_lib.api.resource_sync import (
+    cleanup_after_sync,
+    sync_to_db_without_cleanup,
+)
+from .canvas_helper import to_df
+
+
+STUDENTS_RESOURCE_NAME = "Students"
+
+logger = logging.getLogger(__name__)
+
+
+def students_synced_as_df(
+    students: List[User],
+    sync_db: sqlalchemy.engine.base.Engine,
+) -> DataFrame:
+    """
+    Fetch Students data for a range of courses and
+    return a Students DataFrame
+    with current and previously fetched data
+
+    Parameters
+    ----------
+    students: List[User]
+        a list of Canvas Users objects
+    sync_db: sqlalchemy.engine.base.Engine
+        an Engine instance for creating database connections
+
+    Returns
+    -------
+    DataFrame
+        a Students API DataFrame with the current and previously fetched data
+    """
+    students_df: DataFrame = _sync_without_cleanup(to_df(students), sync_db)
```

Comment:
Comparing the Student list created by the old extractor and your update, there are four extra students now. Apparently we had a bug before that we didn't know about! And now the bug is fixed 😄 .